genienlp/tasks/almond_task.py

```diff
@@ -28,135 +21,309 @@
-            for line in progress_bar(open(path, 'r', encoding='utf-8'), total=max_examples, desc="Reading Dataset"):
-                parts = line.strip().split('\t')
-                examples.append(make_example(parts, dir_name, **kwargs))
-                if len(examples) >= max_examples:
-                    break
-            os.makedirs(os.path.dirname(cache_name), exist_ok=True)
-            if cache_input_data:
-                logger.info(f'Caching data to {cache_name}')
-                torch.save(examples, cache_name)
-
-        super().__init__(examples, **kwargs)
-        
-
-    @classmethod
-    def return_splits(cls, path, train='train', validation='eval', test='test', **kwargs):
-
-        """Create dataset objects for splits of the ThingTalk dataset.
-        Arguments:
-            path: path to directory where data splits reside
-            train: The prefix of the train data. Default: 'train'.
-            validation: The prefix of the validation data. Default: 'eval'.
-            test: The prefix of the test data. Default: 'test'.
-            Remaining keyword arguments: Passed to the splits method of
-                Dataset.
-        """
-        
-        train_data = None if train is None else cls(os.path.join(path, train + '.tsv'), **kwargs)
-        validation_data = None if validation is None else cls(os.path.join(path, validation + '.tsv'), **kwargs)
-        test_data = None if test is None else cls(os.path.join(path, test + '.tsv'), **kwargs)
-
-        aux_data = None
-        do_curriculum = kwargs.get('curriculum', False)
-        if do_curriculum:
-            kwargs.pop('curriculum')
-            aux_data = cls(os.path.join(path, 'aux' + '.tsv'), **kwargs)
-        
-        return Split(train=None if train is None else train_data,
-                     eval=None if validation is None else validation_data,
-                     test=None if test is None else test_data,
-                     aux=None if do_curriculum is None else aux_data)
-    
 
 
 class BaseAlmondTask(BaseTask):
     """Base class for the Almond semantic parsing task
         i.e. natural language to formal language (ThingTalk) mapping"""
-
+    
     def __init__(self, name, args):
         super().__init__(name, args)
+        self.args = args
+        no_feature_fields = ['answer']
+        if self.is_contextual():
+            no_feature_fields.append('context')
+        else:
+            no_feature_fields.append('question')
+        self.no_feature_fields = no_feature_fields
+        
         self._almond_has_multiple_programs = args.almond_has_multiple_programs
         self._almond_detokenize_sentence = args.almond_detokenize_sentence
+        self._almond_thingtalk_version = args.almond_thingtalk_version
+        
+        self.almond_domains = args.almond_domains
+        self.all_schema_types = set()
+
+        if args.do_ner:
+            self.unk_id = args.features_default_val[0]
+            self.TTtype2DBtype = dict()
```

Comment:
As noted by the highlighting, names that start with capital letters are reserved for classes. This is not a class.